_posts/2022-07-01-argocd-image-updater-for-functions.md

```diff
@@ -0,0 +1,248 @@
+
+The **app of apps pattern** is commonly used to install multiple applications in a cluster. Its concept is to declaratively specify one Argo CD app that only consists of other apps. In our case we want to create and deploy two Argo CD apps, one for the OpenFaaS Operator and one for our functions. We will need to create an app called applications that consists of both the openfaas-operator and openfaas-functions app.
+
+The Argo CD Image updater is used to automatically update functions when a new version is published. It polls the image registry to check if a new version of an image is found. If a new version is available and the version constraint is met, the Image Updater instructs Argo CD to update the application with the new image.
+
+![app-of-apps](/images/2022-07-argocd-image-updater/app-of-apps.png)
+> Pictured: The Argo CD UI showing the `applications` app containing both the `openfaas-operator` and `openfaas-functions` app.
+
+### Prerequisites
+
+[arkade](https://github.com/alexellis/arkade) lets developers install the latest versions of their favourite tools and Kubernetes apps. We will use it in this tutorial to install Argo CD and get some other required tools.
+
+Get arkade with:
+```bash
+# Note: you can also run without `sudo` and move the binary yourself
+curl -sLS https://get.arkade.dev | sudo sh
+```
+
+One of the limitations of the Argo CD Image Updater is that you can only update images for applications whose manifest is rendered using either *Kustomize* or *Helm*. The template also needs to support specifying the image name using a parameter. This means that you will need to create a Helm Chart for your functions and make the image names of your functions configurable. We show you how to do this in one of our previous blog post:
+
+- [How to package OpenFaaS functions with Helm](https://www.openfaas.com/blog/howto-package-functions-with-helm/)
+
+### Install ArgoCD
+Create a cluster with KinD if you don't have one already.
+```bash
+kind create cluster
+```
+
+There are various ways to [install ArgoCD](https://argo-cd.readthedocs.io/en/stable/getting_started/#1-install-argo-cd). We are going to use `arkade`
+```
+arkade install argocd
+```
+
+Arkade will print out instructions for port forwarding the Argo CD API and logging in using the ArgoCD CLI. If you need to, you can always retrieve these instructions later by running `arkade info argocd`
+
+We will be using the ArgoCD CLI to manage ArgoCD.
+```bash
+arkade get argocd
+```
+
+Start by port-forwarding the ArgoCD API server to make it accessible on our local host:
+```bash
+kubectl port-forward svc/argocd-server -n argocd 8443:443
+```
+
+Next we will retrieve the password and use it to login with the ArgoCD CLI
+```bash
+# Get the password
+PASS=$(kubectl get secret argocd-initial-admin-secret \
+  -n argocd \
+  -o jsonpath="{.data.password}" | base64 -d)
+
+# Login:
```

Comment:
I think this should be `Log in` as its an instruction / verb, rather than the noun